pkg/apis/workload/v1alpha1/synctarget_types.go

```diff
@@ -94,6 +101,44 @@ type SyncTargetStatus struct {
 	VirtualWorkspaces []VirtualWorkspace `json:"virtualWorkspaces,omitempty"`
 }
 
+type ResourceToSync struct {
+	// group is the group of the API. Empty string for the core API group.
+	//
+	// +required
+	Group string `json:"group"`
+
+	// resource is the resource of the API.
+	//
+	// kubebuilder:validation:MinLength=1
+	// +required
+	Resource string `json:"resource"`
+
+	// Versions is the versions of the resource.
+	// +optional
+	Versions []string `json:"versions"`
```

Comment:
Do we know what this actually means? Does it mean this list is ordered, and the syncer should take the first version available on the cluster and use that for syncing?

Please update go doc.

Reply:
change to `Version string`, I think we need to get state for each GVR from syncer.

Reply:
but we have this problem of skew. Old kube might only have beta version of some API, newer kube has v1 and beta. Newest kube has only v1. We need a plan here how to cope with that. cc @davidfestal

Reply:
and this poses the question who will do the conversion. 

Side-note: @ncdc, @stevekuznetsov and I talked about CEL conversions yesterday for APIResourceSchemas. Video in slack.

Reply:
Proposal:
```golang
// versions are the resource versions the syncer can choose to sync depending on
// availability on the downstream cluster. Conversion to the storage version, if necessary, 
// will be done on the kcp side. The versions are ordered according by precedence, i.e. the
// first version is preferred.
//
// +kubebuilder .... MinItems=1
// +required
Versions []string `json:"versions"`
```